xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -236,9 +236,12 @@ bool CGUIMediaWindow::OnMessage(CGUIMessage& message)
       m_iSelectedItem = m_viewControl.GetSelectedItem();
       m_iLastControl = GetFocusedControlID();
       CGUIWindow::OnMessage(message);
-      CGUIDialogContextMenu* pDlg = (CGUIDialogContextMenu*)g_windowManager.GetWindow(WINDOW_DIALOG_CONTEXT_MENU);
-      if (pDlg && pDlg->IsActive())
-        pDlg->Close();
+
+      // close related dialogs
+      CGUIDialog* pDialog;
+      pDialog = (CGUIDialog*)g_windowManager.GetWindow(WINDOW_DIALOG_CONTEXT_MENU);
+      if (pDialog && pDialog->IsActive())
+        pDialog->Close();
```

Comment:
IMO we should do the same as we do for all the other dialogs, check if running not active and force close.

```
if (pDialog && pDialog->IsDialogRunning())
  pDialog->Close(true);
```